src/app/shared/records/services/records.service.ts

```diff
@@ -84,50 +84,50 @@ export class RecordsService {
   }
 
   public get(): Observable<any> {
-    if (this.isTraineesState) {
-      return this.store.dispatch(new GetTrainees());
-    } else if (this.isConcernsState) {
-      return this.store.dispatch(new GetConcerns());
+    if (!this.getAction) {
+      throw new Error("getAction must be defined");
     }
+
+    return this.store.dispatch(new this.getAction());
   }
 
-  public sort(active: string, direction): void {
-    if (this.isTraineesState) {
-      this.store.dispatch(new SortTrainees(active, direction));
-    } else if (this.isConcernsState) {
-      this.store.dispatch(new SortConcerns(active, direction));
+  public sort(active: string, direction): Observable<any> {
+    if (!this.sortAction) {
+      throw new Error("sortActionFunction must be defined");
```

Comment:
Could you please explain this? Why do we want to throw error?
What will happen in the page?